epsagon/trace.py

```diff
@@ -112,30 +112,79 @@ def get_trace(self):
                 self.debug
             )
             self.traces[thread_id] = new_trace
-
         return self.traces[thread_id]
 
+    def get_trace(self):
+        """
+        Get the trace of the current thread.
+        :return:
+        """
+        return self.traces.get(threading.currentThread().ident)
+
     def remove_trace(self):
         """
         Remove the thread's trace.
         """
         self.traces.pop(threading.currentThread().ident, None)
 
+    def add_event(self, event):
+        """
+        Add  event to the relevant trace.
+        :param event: The event to add.
+        :return: None
+        """
+        if threading.currentThread().ident in self.traces:
+            self.get_trace().add_event(event)
+
+    def add_exception(self, exception, stack_trace, additional_data=''):
+        """
+        add an exception to the relevant trace.
+        :param exception: the exception to add
+        :param stack_trace: the traceback at the moment of the event
+        :param additional_data: a json serializable object that contains
+            additional data regarding the exception
+        :return: None
+        """
+        if threading.currentThread().ident in self.traces:
```

Comment:
It was better not to duplicate this line - you could add a separate function and call it each time (even with a decorator :P)

Reply:
Yeah thought about it already lol :/